symbology/symbol-dictionary-sample/src/main/java/com/esri/samples/symbology/SymbolDictionaryController.java

```diff
@@ -0,0 +1,173 @@
+  @FXML private TextField nameField;
+  @FXML private TextField tagField;
+  @FXML private TextField symbolClassField;
+  @FXML private TextField categoryField;
+  @FXML private TextField keyField;
+  @FXML private Text searchResultsFound;
+  @FXML private ProgressIndicator progress;
+
+  private VBox searchResultsBox;
+
+  private SymbolDictionary dictionarySymbol;
+  private StyleSymbolSearchParameters searchParameters;
+
+  /**
+   * 
+   */
+  public void initialize() {
+    // loads a specification for the symbol dictionary
+    dictionarySymbol = new SymbolDictionary("mil2525d");
+    dictionarySymbol.loadAsync();
+
+    // parameters used to search for symbol dictionary
+    searchParameters = new StyleSymbolSearchParameters();
+  }
+
+  /**
+   * Searches through the symbol dictionary using the text from the search fields.
+   * 
+   * @param event action from search button
+   */
+  @FXML
+  private void handleSearchAction(ActionEvent event) {
+    // accessing text from all search fields 
+    searchParameters.getNames().add(nameField.getText());
+    searchParameters.getTags().add(tagField.getText());
+    searchParameters.getSymbolClasses().add(symbolClassField.getText());
+    searchParameters.getCategories().add(categoryField.getText());
+    searchParameters.getKeys().add(keyField.getText());
+
+    new Thread(() -> {
+      try {
+        progress.setVisible(true);
+        // searching for any matches in dictionary
+        ListenableFuture<List<StyleSymbolSearchResult>> searchResult =
+            dictionarySymbol.searchSymbolsAsync(searchParameters);
+        List<StyleSymbolSearchResult> symbolResults = searchResult.get();
+
+        // displays number of results
+        Platform.runLater(() -> searchResultsFound.setText("" + symbolResults.size()));
+        //        searchResultsFound.setText("" + symbolResults.size());
+
+        // reset result so duplicates are not added
+        searchResultsBox = new VBox(10);
+        if (symbolResults.size() > 0) {
+          symbolResults.forEach(e -> {
+            new Thread(() -> {
+              displaySymbol(e);
+            }).start();
+          });
+        } else {
+          Text description = new Text("No Results Found");
+          searchResultsBox.getChildren().add(description);
+        }
+        Platform.runLater(() -> scrollPane.setContent(searchResultsBox));
+      } catch (ExecutionException ee) {
+        System.out.println("Dictionary Symbol search was aborted!\n" + ee.getMessage());
+      } catch (InterruptedException ie) {
+        System.out.println("Thread interrupted during Dictionary Symbol search!\n" + ie.getMessage());
+      } finally {
+        progress.setVisible(false);
+      }
+    }).start();
+  }
+
+  /**
+   * Displays an image of the symbol passed along with its full name, tags, symbol class, category, and key.
+   * 
+   * @param symbolResult symbol to be displayed
+   */
+  private void displaySymbol(StyleSymbolSearchResult symbolResult) {

```

Comment:
This symbol view should be a custom component with its own fxml file since its handling mostly "view" logic.